### DISCUSSION TITLE
internal server error for validation errors

### DISCUSSION THREAD

**Discussion by nels0n12** [adonisjs/core#4084] [Help]
### Package version

╭────────────────────────────────────╮
│    node ace --version                                                           
│────────────────────────────────────
│                                                                                             
│    App version: 1.0.0                                                            
│    Framework version: 5.9.0                                                 
│    Assembler version: 5.9.5                                                              
│                                    
╰────────────────────────────────────╯

### Node.js and npm version

Node: v16.17.0 / npm: v8.15.0 (yarn v1.22.19)

validation returns a 500 internal server error. not sure why this is happening. 

this is my code 
```
const newPostSchema = schema.create({
        id_no: schema.string({}, [
          rules.escape(),
          rules.trim(),
          rules.minLength(9),
          rules.maxLength(9),
        ]),
      })
const payload = await request.validate({
        schema: newPostSchema,
        reporter: validator.reporters.api,
      })
```
import = `import { rules, schema, validator } from '@ioc:Adonis/Core/Validator'`

below is the image of the error
![image](https://user-images.githubusercontent.com/55479584/221135929-259f5a01-f7f6-4575-b914-f92f5c511709.png)

and the logs on the terminal are shown below
![image](https://user-images.githubusercontent.com/55479584/221136391-f1aa3d6f-2152-4a7e-b40e-37485e49e857.png)

**Comment by RomainLanz** [comment]
Hey @nels0n12! 👋🏻 

What your global `ExceptionHandler` looks like?

**Reply by nels0n12** [reply]
Hi  my ExceptionHander is as below
```
import Logger from '@ioc:Adonis/Core/Logger'
import HttpExceptionHandler from '@ioc:Adonis/Core/HttpExceptionHandler'
import { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'

export default class ExceptionHandler extends HttpExceptionHandler {
  constructor() {
    super(Logger)
  }

  public async handle(error: any, ctx: HttpContextContract) {
    /**
     * Self handle the validation exception
     */
    if (error.code === 'E_VALIDATION_FAILURE') {
      return ctx.response.status(422).json({
        success: false,
        message: 'E_VALIDATION_FAILURE',
        data: error.messages,
      })
    }

    /**
     * Forward rest of the exceptions to the parent class
     */
    return super.handle(error, ctx)
  }
}
```

even before I edited the exception I still got the same error and it was as follows

```
import Logger from '@ioc:Adonis/Core/Logger'
import HttpExceptionHandler from '@ioc:Adonis/Core/HttpExceptionHandler'

export default class ExceptionHandler extends HttpExceptionHandler {
  constructor() {
    super(Logger)
  }
}
```

**Reply by nels0n12** [reply]
Hi @RomainLanz 

can you please suggest a solution to this issue
